benchmarks/src/main/java/io/grpc/benchmarks/Utils.java

```diff
@@ -220,6 +224,38 @@ public static ManagedChannel newClientChannel(Transport transport, SocketAddress
     }
     if (directExecutor) {
       builder.directExecutor();
+    } else {
+      /*
+       * When running the higher QPS (>10000) benchmarks, there can sometimes be bad behavior using
+       * the default channel executor.  To address this, we use a fixed sized thread pool, which is
+       * specially made to not have contention on the internal queue of Runnables.  This allows for
+       * better scaling.
+       *
+       * Since we aren't using the blocking API, we pick the parallelism to be equal to the number
+       * of available processors.  We also know that the benchmark client and server are running on
+       * separate machines so we have effectively full control over all the processors.
+       * Additionally, we don't ever join on the ForkJoin tasks, so we run in async mode.
+       *
+       * In order to make sure threads have the right name and don't block shutdown, we use a custom
+       * thread factory.
+       *
+       * Only make these changes if you are certain they are right for you.  We chose them to use
+       * them here after careful testing and benchmarking.
+       *
+       * See https://github.com/grpc/grpc-java/issues/2118 for more details.
+       */
+      builder.executor(new ForkJoinPool(Runtime.getRuntime().availableProcessors(),
```

Comment:
Since this is already in a util class, should we add a new method that would create the executor? It could then be used in the three different places and we wouldn't need to copy/paste multiple paragraphs worth of docs.